src/components/TodoInput.tsx

```diff
@@ -0,0 +1,100 @@
+import { ChangeEvent, useContext, useState } from 'react';
+import { TodoContext } from './TodoContext';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {};
+
+function getNewId(todos: Todo[]) {
+  if (todos.length === 0) {
+    return 1;
+  }
+
+  let maxId = 0;
+
+  for (let i = 0; i < todos.length; i++) {
+    if (maxId <= todos[i].id) {
+      maxId = todos[i].id;
+    }
+  }
+
+  return maxId + 1;
+}
+
+export const TodoInput: React.FC<Props> = () => {
+  const { todos, setTodos } = useContext(TodoContext);
+  const [title, setTitle] = useState('');
+  const [isFocused, setIsFocused] = useState(true);
+
+  const handleCompletedAll = () => {
+    const isCompleted = todos.every(todo => todo.completed);
+
+    if (!isCompleted) {
+      const newTodos = todos.map(todo => ({
+        ...todo,
+        completed: true,
+      }));
+
+      setTodos(newTodos);
+    } else {
+      const newTodos = todos.map(todo => ({
+        ...todo,
+        completed: false,
+      }));
```

Comment:
```suggestion

      const newTodos = todos.map(todo => ({
        ...todo,
        completed: !isCompleted,
      }));

      setTodos(newTodos);
```
looks like we can simplify it